analytics.py

```diff
@@ -30,5 +37,5 @@ def __init__(self, _db):
     def collect_event(self, library, license_pool, event_type, time=None, **kwargs):
         if not time:
             time = datetime.datetime.utcnow()
-        for provider in self.providers:
+        for provider in (self.sitewide_providers + self.library_providers[library.id]):
```

Comment:
When you call collect_event() to register the same event for multiple libraries, will this cause the same event to be registered multiple times for the sitewide providers?

If so, it might work better for the sitewide providers to be handled as providers that, if enabled, are presumed to be enabled for every library in the system.